src/TSTransformer/nodes/expressions/transformIdentifier.ts

```diff
@@ -129,7 +130,14 @@ export function transformIdentifier(state: TransformState, node: ts.Identifier)
 	const constructSymbol = getFirstConstructSymbol(state, node);
 	if (constructSymbol) {
 		const constructorMacro = state.services.macroManager.getConstructorMacro(constructSymbol);
-		if (constructorMacro) {
+		if (
+			constructorMacro &&
+			// Inhibit if extending from this identifier - noMacroExtends diagnostic will catch it
+			!(
+				ts.isClassLike(node.parent.parent.parent) &&
+				getExtendsNode(node.parent.parent.parent)?.expression === node
+			)
```

Comment:
I think it's more accurate to emit both diagnostics.

Reply:
The user solving `noMacroExtends` will always solve `noConstructorMacroWithoutNew`. I don't see what the added value of also reporting the latter would be; If anything, it's shown to be confusing in the past because the user doesn't think the `extends` reference _is_ a constructor.

Reply:
I don't like the idea of this code assuming that the other code exists and ran.

If we ever remove `noMacroExtends` for some reason, we could easily forget to remove this check.